simulator/cctree.cpp

```diff
@@ -44,9 +44,11 @@ void CCTree::handle_object_allocation(int object_id, int size, std::string type,
     }
   }
   last_thread_id = thread_id;
+
+  return heapObject;
 }
 
-void CCTree::handle_object_death(int object_id) {
+HeapObject * CCTree::handle_object_death(int object_id) {
```

Comment:
Should a pointer be returned here?

Reply:
This is pretty much dead code because death events are deprecated. This was changed to return the heap object that died so the simulator can use that information somehow. Should I delete this?

Reply:
Not necessarily. If the information about the HeapObject is necessary, then it's probably good to keep. I'm just making a note that returning a raw pointer is dangerous. It means that we don't know who owns it (e.g. who is supposed to free the object). It's a good idea to comment who owns the raw pointer (the caller, callee, or both) but even more preferable to use a smart pointer and have the code do it for you. These are nice blog posts that explain them: https://herbsutter.com/2013/05/29/gotw-89-solution-smart-pointers/ https://herbsutter.com/2013/05/30/gotw-90-solution-factories/

Reply:
Ok fair enough. I changed the return value to void.